### DISCUSSION TITLE
Run task on specific host - v7

### DISCUSSION THREAD

**Discussion by mvuidev** [deployphp/deployer#2954] [Help needed]
Hello everyone,

I'm trying to assign a specific host to my tasks with "select()" but it's not working.
For example : 

```php
localhost()->set('labels', ['host' => 'local']);
host('myhost')->set('labels', ['host' => 'remote']);

task('test', function(){
    // My task
})->select('host=local');
```

It keeps asking me to select hosts.
Any idea ? Am I using it correctly ?

PHP 7.4
Deployer 7.0.0-rc4

**Comment by antonmedv** [comment]
What command do you trying to run?

try dep deploy all

**Reply by mvuidev** [reply]
Hi @antonmedv ,

With my example (not using deploy tasks), I run "dep test" and expect it to run on the host "local".
If I run "dep test all" it runs on the two hosts.

Is my example supposed to work if I run "dep test" or I'm missing something ?

**Reply by antonmedv** [reply]
Yes, dep command does not determine what hosts to select. It's provided by the user.

**Reply by mvuidev** [reply]
Yeah seems like I need to run "dep test host=local".
But in this case what's the purpose of ->select() ?
Is there currently no way to assign a host to a task ?

Trying to understand 😄 
And, thanks for this great tool by the way 👍

**Reply by antonmedv** [reply]
It’s more like a filter. Which hosts will be used for this task .

**Reply by mvuidev** [reply]
Okay, I understand now.

Just one thing : still about my example, if I run "dep test remote", the task is not executed (because of the host=local) but it never ends, have to ctrl+c to stop it.

**Reply by antonmedv** [reply]
Seems like a bug. Please provide full example.

**Reply by mvuidev** [reply]
Here is a simple example :

```php
<?php

namespace Deployer;

localhost('local');
host('remote')->setHostname('myhost');

task('mytask', function ()
{
    writeln('Hello');
})->select('alias=remote');
```

Run it :

```
$ dep mytask local -vvv
task mytask
done mytask 1ms
|
```
Have to ctrl+c to stop it.

**Reply by antonmedv** [reply]
Please, run with -vvv.

**Reply by mvuidev** [reply]
I did in my example : 

```
$ dep mytask local -vvv
```

**Comment by jonnitto** [comment]
I have to say, that I think it would be useful to set a task to run local only. Sometimes you have multiple hosts, but want to run a local task only...

**Reply by antonmedv** [reply]
runLocally()?

**Reply by jonnitto** [reply]
runLocally is inside a task, but I would like to have something like taskLocally, or `task()->local()` If I have a local task, it still connects to the server. And therebefore, if you have multiple servers, you get asked which server the task should run. In v6 I could solve this with `->once()`, which was kind of a bug, that you wouldn't get asked for the server. but sometimes it exactly what you want

**Reply by antonmedv** [reply]
Hmm, this is actually a good point. Local task doesn't need a host to connect to. So basically you use v6 local() as task runner?

**Reply by antonmedv** [reply]
Can you show an example?

**Reply by jonnitto** [reply]
As an example I use it here:
https://github.com/jonnitto/Uberspace.Deployer/blob/main/tasks/git.php
